pyk/src/pyk/proof/reachability.py

```diff
@@ -790,20 +829,52 @@ def step_proof(self, step: APRProofStep) -> list[APRProofResult]:
 
         # Ensure that we record progress ASAP for circularities, so the circularity rule will be included for execution as soon as possible
         execute_depth = self.execute_depth
-        if step.circularity and not step.nonzero_depth and (execute_depth is None or execute_depth > 1):
+        if step.circularity and not step.nonzero_depth:
             execute_depth = 1
 
-        extend_result = self.kcfg_explore.extend_cterm(
-            step.node.cterm,
-            execute_depth=execute_depth,
-            cut_point_rules=cut_rules,
-            terminal_rules=self.terminal_rules,
-            module_name=step.module_name,
-            node_id=step.node.id,
-        )
+        # If the step has already been cached, do not invoke the backend and only send a signal back to the proof to use the cache
+        if step.use_cache is not None:
```

Comment:
What happens if the prover is terminated with cached results not yet applied? Is it still possible to continue extending the KCFG?

Reply:
Yes, it is, and cached results should be re-used if the proof has survived. If they are lost, there is no loss for the KCFG, since they are only a snapshot of guaranteed future extension results.